### DISCUSSION TITLE
GNU compiler problem

### DISCUSSION THREAD

**Discussion by nyYstanton** [MetOffice/simulation-systems#172] [UM]
Installed UM under CentOS7 with GNU compiler devtoolset 7.3.1

Recently our system upgraded to CentOS8 and we have to use gnu compiler over 8 or 9.

Before gnu version 7.3.1, we don't have any problem to simulate so long (more than 30 years), but now I cannot do simulate UM more than 7 months. 

Do you kindly suggest what should I do properly if we were persuaded to long-term simulate under GCC version over 8?

Here are my software and module lists

GNU Compiler (C, C++, Fortran) 9.2.1
MPICH 3.1.4
HDF5 1.10.3
NetCDF-C 4.7.4
NetCDF-Fortrann 4.5.0
GCOM 8.0
Shumlib 4851

**Comment by richardgilham** [comment]
Hi there,

You can see what compiler versions other sites are running at here with plenty of examples running at later versions of GNU: https://code.metoffice.gov.uk/trac/um/wiki/Compilers

Is your issue that the UM fails after 7 months, or that it's significantly slower?

Also, it would be worth checking that your GCOM and SHUMlib versions are aligned with the UM version you're using.

**Reply by nyYstanton** [reply]
Thanks for replying Sir and also sorry for my poor english.
My issue is that UM fails after 5-to-7 months simulations with this log messages. 
And I think I don't show you the log messages so I put them here

#0  0x145b4dc423ff in ???
#1  0x824ba2 in icb_lbc_mpp_nfld
        at /home/vagrant/cylc-run/u-bp039_coupled_cmip6_stratrop/share/fcm_make_ocean/preprocess-ocean/src/nemo/NEMOGCM/NEMO/OPA_SRC/ICB/icblbc.F90:704
#2  0x828daa in __icblbc_MOD_icb_lbc_mpp
        at /home/vagrant/cylc-run/u-bp039_coupled_cmip6_stratrop/share/fcm_make_ocean/preprocess-ocean/src/nemo/NEMOGCM/NEMO/OPA_SRC/ICB/icblbc.F90:543
#3  0x80faf8 in __icbstp_MOD_icb_stp
        at /home/vagrant/cylc-run/u-bp039_coupled_cmip6_stratrop/share/fcm_make_ocean/preprocess-ocean/src/nemo/NEMOGCM/NEMO/OPA_SRC/ICB/icbstp.F90:120
#4  0x568cf1 in __sbcmod_MOD_sbc
        at /home/vagrant/cylc-run/u-bp039_coupled_cmip6_stratrop/share/fcm_make_ocean/preprocess-ocean/src/nemo/NEMOGCM/NEMO/OPA_SRC/SBC/sbcmod.F90:440
#5  0x8f4342 in __step_MOD_stp
        at /home/vagrant/cylc-run/u-bp039_coupled_cmip6_stratrop/share/fcm_make_ocean/preprocess-ocean/src/nemo/NEMOGCM/NEMO/OPA_SRC/step.F90:138
#6  0x4185e7 in __nemogcm_MOD_nemo_gcm
        at /home/vagrant/cylc-run/u-bp039_coupled_cmip6_stratrop/share/fcm_make_ocean/preprocess-ocean/src/nemo/NEMOGCM/NEMO/OPA_SRC/nemogcm.F90:152
#7  0x414cdf in nemo
        at /home/vagrant/cylc-run/u-bp039_coupled_cmip6_stratrop/share/fcm_make_ocean/extract/nemo/NEMOGCM/NEMO/OPA_SRC/nemo.f90:18
#8  0x414d16 in main
        at /home/vagrant/cylc-run/u-bp039_coupled_cmip6_stratrop/share/fcm_make_ocean/extract/nemo/NEMOGCM/NEMO/OPA_SRC/nemo.f90:11

I still getting this error messages and I honestly don't figure them out yet.

**Comment by richardgilham** [comment]
Thanks for the information. I've asked some colleagues and there's not much we can do from here. Which institution are you at? There might be able to suggest someone near that can help?
